### PR TITLE
Migrate to Spago in preparation for publishing in package-sets

### PR DESCRIPTION
**Pull request by fsoikin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+    "email": "daniel@gasienica.ch",
+    "url": "https://www.gasi.ch"
+  },
   "contributors": [
-    "Daniel Gasienica <daniel@gasienica.ch>"
+    {
+      "name": "Fyodor Soikin",
+      "email": "fyodor@collegevine.com",
+      "url": "https://github.com/fsoikin"
+    }
```
Comment (fsoikin) [inline package.json:15]: Note here: I changed these fields to conform them to the official `package.json` schema and used your details from https://github.com/collegevine/purescript-elmish/pull/26 @gasi

File: packages.dhall [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      https://github.com/purescript/package-sets/releases/download/psc-0.13.8-20201222/packages.dhall sha256:620d0e4090cf1216b3bcbe7dd070b981a9f5578c38e810bbd71ece1794bfe13b
+
+let overrides = {=}
+
+let additions =
+      { elmish =
+          https://raw.githubusercontent.com/collegevine/purescript-elmish/master/spago.dhall sha256:79f4da7bb6e1fa837767e2532dee1e4b162ba660748ef4070f4ca404aa1dcd01
+        with repo = "https://github.com/collegevine/purescript-elmish"
+        with version = "v0.3.1"
+      }
```
Comment (fsoikin) [inline packages.dhall:11]: This is extra-depped here for now, because the new package set containing Elmish hasn't been released yet. Will change this to the new package set version after publishing.

File: src/Elmish/HTML.purs [hunk 3]
```diff
@@ -4,6 +4,6 @@ module Elmish.HTML
     , module ExportedInternal
     ) where
 
-import Elmish.React.DOM (text) as ExportedDOM
+import Elmish.React.DOM (text, empty, fragment) as ExportedDOM
```
Comment (fsoikin) [inline src/Elmish/HTML.purs:7]: Added reexports of `empty` and `fragment` as a drive-by improvement - something I wanted to do for a while.

File: src/Elmish/HTML/Styled.purs [hunk 4]
```diff
@@ -34,6 +34,6 @@ module Elmish.HTML.Styled
     , module ExportedInternal
     ) where
 
-import Elmish.React.DOM (text) as ExportedDOM
+import Elmish.React.DOM (text, empty, fragment) as ExportedDOM
```
Comment (fsoikin) [inline src/Elmish/HTML/Styled.purs:37]: Ditto

**Review by mcordova47 — APPROVED** [review]
